server/manager/src/main/java/org/apache/accumulo/manager/compaction/queue/CompactionJobPriorityQueue.java

```diff
@@ -138,6 +150,10 @@ public CompactionJobPriorityQueue(CompactorGroupId groupId, int maxSize) {
     this.rejectedJobs = new AtomicLong(0);
     this.dequeuedJobs = new AtomicLong(0);
     this.futures = new ArrayDeque<>();
+    this.jobAges = new ConcurrentHashMap<>();
```

Comment:
This could be a HashMap because all accesses to it should be inside a sync block.

Reply:
I made it a ConcurrentHashMap() because of CompactionJobPriorityQueueStats iterates over jobAges to build the stats. It's memoized but it's accessible publically so I was thinking another thread could call to get the metrics and you could end up iterating over the map to build the stats outside the lock.

Reply:
> I made it a ConcurrentHashMap() because of CompactionJobPriorityQueueStats iterates over jobAges to build the stats.

ok, that is a good reason for it to be a ConcurrentHashMap